### PR TITLE
Add patch option in bundle config

### PR DESCRIPTION
**Pull request by ankitkataria**
### What was the end-user problem that led to this PR?

Issue #5994 

### What was your diagnosis of the problem?

As mentioned by @indirect I added a `patch` option in the bundler config, which by default is false. If set to true, patch level update are preferred if no update levels are specified. 

### What is your fix for the problem, implemented in this PR?

A patch level config option which can be opted into.

### CODE REVIEW

File: spec/commands/update_spec.rb [hunk 1]
```diff
@@ -809,6 +809,15 @@
       G
     end
 
+    context "with patch set as default update level in config" do
+      it "should do a patch level update" do
+        bundle! "config --local patch true"
+        bundle! "update --patch foo"
```
Comment (greysteil) [inline spec/commands/update_spec.rb]: This shouldn't include the `--patch`, right?

```suggestion
        bundle! "update foo"
```

File: man/bundle-config.ronn [hunk 2]
```diff
@@ -220,6 +220,8 @@ learn more about their operation in [bundle install(1)](bundle-install.1.html).
 * `only_update_to_newer_versions` (`BUNDLE_ONLY_UPDATE_TO_NEWER_VERSIONS`):
    During `bundle update`, only resolve to newer versions of the gems in the
    lockfile.
+* `patch` (BUNDLE_PATCH):
+   Whether to use patch update level, by default set to false.
```
Comment (greysteil) [inline man/bundle-config.ronn]: Coming to the conversation fresh, this description doesn't give me enough information. Perhaps the option could be renamed to `prefer_patch`, and the description could read something like:

```
Prefer updating only to next patch version during updates. Makes `bundle update` calls equivalent to `bundler update --patch`.
```

File: spec/commands/update_spec.rb (hunk 1 above)
Comment (ankitkataria) [reply spec/commands/update_spec.rb]: Yes, corrected that. :sweat_smile:

File: man/bundle-config.ronn (hunk 2 above)
Comment (ankitkataria) [reply man/bundle-config.ronn]: Yes, prefer_patch seems to be a better and the text you've given makes much more sense to a new user. I've updated that as well.

**Comment by greysteil** [comment]
@chrismo are you also 👍 ?

**Comment by indirect** [comment]
@bundlerbot r=greysteil :)

**Pull request merged**